### DISCUSSION TITLE
Stop throwing Error 406 when fetching Single Row - Supabase

### DISCUSSION THREAD

**Discussion by Marcus20119** [supabase/supabase#12177] [Questions]
Hi guys, I used this to check if my user liked the movie or not:

```
const { data, error, status } = await supabase
   .from(favorite_movies)
   .select()
   .match({ user_id: session.user.id, movie_id: movie_id })
   .single();
```

Everything is totally okay but when my user has not liked the movie yet (the table 'favorite_movies' does not contain the row I wanna fetch), supabase will throw an Error with code 406 and expose my userId on the console. Something like this: 
```
GET https://fnpwntjafdaszvfesblb.supabase.co/rest/v1/favorite_movies?select=*&user_id=eq.9f69bfed-63c3-4077-803f-2fe42a68a8f7&movie_id=eq.130542 406
```

I want Supabase to stop throwing this Error or at least hide it from the console, what should I do?

**Comment by GaryAustin1** [comment]
Then don't use single and just check if data array length returned is 0.
I'm also not sure why your user id being exposed is of concern.  It is in local storage and visible in the network tab.  Knowing the uuid is not a security risk if you have RLS setup.
